ftl/cached/cached.py

```diff
@@ -78,17 +88,14 @@ def _fetch_lyr_shas(self, img_name):
                 lyr_shas.append(lyr['digest'])
             return set(lyr_shas)
 
-    def _compare_layers(self, lyr_shas_1, lyr_shas_2):
-        if len(lyr_shas_1) != len(lyr_shas_2):
-            raise RuntimeError("different amount of layers found when \
-                               reuploading same image")
-        lyr_diff_cnt = 0
-        for lyr_sha in lyr_shas_1:
-            if lyr_sha not in lyr_shas_2:
-                lyr_diff_cnt += 1
-            if lyr_diff_cnt > 1:
-                raise RuntimeError("more layers differed then app layer when \
-                    reuploading image")
+    def _compare_layers(self, lyr_shas_1, lyr_shas_2, offset):
+        lyr_diff = lyr_shas_1 - lyr_shas_2
+        logging.info(
+            "Encountered %s differences between layers" % len(lyr_diff))
+        logging.info("Different layer shas: %s" % lyr_diff)
+        if len(lyr_diff) != offset:
+            raise RuntimeError("expected {0} different layers, got {1}".format(
```

Comment:
I'm not a huge fan of raising a generic `RuntimeError` since that can get thrown by a lot of different things, if you're gonna do this pattern can you make it throw a custom exception? You can pretty quickly define something like `FTLException` in the utils module and then use that everywhere.

Reply:
done